### PR TITLE
Oauth: Implement signInWithOAuth Mutation Resolver

### LINKED ISSUE #4385: Oauth - Implement signInWithOAuth Mutation Resolver
## 1. Problem
Users must be able to sign in (or register) via OAuth providers. We need a GraphQL mutation to exchange the authorization code, fetch profile, find/link/create user, and return AuthenticationPayload.

## 2. Scope
- Implement signInWithOAuth(input: OAuthLoginInput!): AuthenticationPayload!
- Handle three paths: existing linked account, email match -> link, no match -> create user + link
- Update lastUsedAt on successful sign-in

## 3. Approach
- Use provider from context.getOAuthProvider(input.provider)
- Exchange code -> access token
- Fetch profile -> providerId, email, name, picture
- Drizzle: find oauth_accounts by (provider, providerId)
- If not found: try find user by email; link or create; insert into oauth_accounts
- Generate JWT using existing utilities; return tokens + user

## 4. Dependencies
- **Blocked By**: #4379, #4380, #4382, #4383, #4384
- **Blocks**: None

## 5. Files to Modify/Create
- src/resolvers/Mutation/signInWithOAuth.ts (create)
- src/resolvers/Mutation/index.ts (modify — export)
- src/utilities/auth/jwt.ts (read-only use)

## 6. Acceptance Criteria
- All existing functionality must be maintained
- Implementation must ensure no breaking changes when PR is merged
- Correct linking/creation logic; lastUsedAt updated
- Returns AuthenticationPayload consistent with existing signIn

## 7. Testing Requirements
- All tests must pass successfully
- Code coverage must be >= 96%
- Unit tests for each path (existing, link, create)
- Error-path tests (invalid code, provider errors)

## 8. Sample Code
**Disclaimer**: This sample code is only representative of what could be done. It is provided only to give a general outline. You will need to use your initiative and industry best practices to find a solution that is suitable for the application.

## 4. Dependencies (Updated)
- **Blocked By**: #4379, #4380, #4382, #4383, #4384
- **Blocks**: None

### PR DESCRIPTION
**Pull request by Jashan32**
**What kind of change does this PR introduce?**
Implemented signInWithOAuth Mutation Resolver

**Issue Number:** #4385

Fixes #4385

**Snapshots/Videos:**
image

**If relevant, did you update the documentation?**
relevant docs already updated

**Summary**
Implemented signInWithOAuth Mutation Resolver

**Does this PR introduce a breaking change?**
NO

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

## Summary by CodeRabbit

* **New Features**
  * Full OAuth sign-in: exchanges auth code, verifies profile, links accounts (existing link, matching verified email, or new user), issues access and refresh tokens, and optionally sets an HTTP-only cookie. Returns user role and email-verified status.

* **Bug Fixes / Reliability**
  * Granular error responses, transactional linking, resets failed-login state, and updates last-used timestamps.

* **Tests**
  * Comprehensive integration-style suite covering successful flows, linking cases, role upgrades, cookie behavior, and many error/edge cases.

### CODE REVIEW

**Comment by Jashan32** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where mock cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

**Comment by Jashan32** [comment]
@coderabbitai The issue is that the database has a foreign key constraint preventing the creation of OAuth accounts with non-existent user IDs.

**Comment by Jashan32** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where mock cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

**Comment by Jashan32** [comment]
@palisadoes PR is ready for review

File: src/graphql/types/Mutation/signInWithOAuth.ts [hunk 1]
```diff
@@ -20,13 +40,373 @@ builder.mutationField("signInWithOAuth", (t) =>
...
+
+				// Set HTTP-Only cookies for web clients if cookie helper is available
+				if (ctx.cookie) {
+					ctx.cookie.setAuthCookies(accessToken, rawRefreshToken);
+				}
+
+				return {
+					authenticationToken: accessToken,
+					refreshToken: rawRefreshToken,
+					user,
```
[the remark this answers is not in the document]
Comment (Jashan32) [reply src/graphql/types/Mutation/signInWithOAuth.ts:408]: authentication context is only used after successful token generation and response completion, so no partial auth state can be observed.
